actors/verifreg/src/lib.rs

```diff
@@ -441,8 +447,10 @@ impl Actor {
                     format!("failed to remove allocation {}", claim_alloc.allocation_id),
                 )?;
 
-                datacap_claimed += DataCap::from(claim_alloc.size.0);
+                total_datacap_claimed += DataCap::from(claim_alloc.size.0);
                 ret_gen.add_success();
+                sector_claims
+                    .push(SectorAllocationClaimResult { claimed_space: claim_alloc.size.0.into() });
```

Comment:
I might be missing something here, but wouldn't `map` over params.allocations be sufficient to ensure that the result array stays parallel?

Also, given `BatchReturn` is no longer part of `ClaimAllocationsReturn`, I'll remove the `ret_gen` tracking, though this drops the nuance of specific error codes from the error message (it's already missing from the return value now).

Reply:
I don't know if [this](https://github.com/filecoin-project/builtin-actors/pull/1304/commits/23f7f5644946ea8ef0f0fb5f1e4d751d78b98c60) is a valid. Is it possible to have a valid but zero-sized claim?

Reply:
I think `map` is hard to use if you want to bubble a result with `?`, like at 433 (unless you want to deal with a `Map<Result<>>`.  That's why `try_map`.

This logic of checking for zero size is valid at the moment, because the minimum Allocation size is positive. It's probably safe forever, just nervous.